tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java

```diff
@@ -136,6 +138,52 @@
         options.addOption(d);
     }
 
+    /**
+     * Reads the attributes that are required to be preserved in the file.
+     * The attributes are posix, dos and acl ones that are interested to
+     * maintain.
+     *
+     * @param path The path to get the attributes from
+     * @return The map of attributes (not null) with keys prefixed with the type (e.g. "posix:permissions")
+     * @throws IOException Some error reading the file attributes
+     */
+    public static Map<String, Object> readAttributesForPreservation(Path path) throws IOException {
+        Map<String, Object> attrs = new HashMap<>();
+        if (Files.exists(path)) {
+            // copy all the permissions that are going to be maintained (posix, dos and acl)
+            Set<String> supportedViews = path.getFileSystem().supportedFileAttributeViews();
+            if (supportedViews.contains("posix")) {
+                attrs.putAll(Files.readAttributes(path, "posix:permissions")
+                        .entrySet().stream().collect(Collectors.toMap(e -> "posix:" + e.getKey(), e -> e.getValue())));
```

Comment:
For code like this please use traditional for loops / iterators etc to iterate the values in the collection.

This is a command line tool so not a big deal in this case but we try and avoid using streams for collection iteration as they cause a number of classes to be defined / instantiated contributing to the overall overhead of the application server.

Reply:
Done! I have created a helper method to add all the attributes with the prefix using a common loop.